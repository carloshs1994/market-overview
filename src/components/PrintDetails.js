import React from 'react';
import PropTypes from 'prop-types';

const PrintDetails = ({ stockDetails }) => Object.entries(stockDetails).map(([id, value]) => (
  <li key={id}>
    <p>{id}</p>
    <p>{(value !== null) ? value : 'N/A'}</p>
  </li>
));

PrintDetails.propTypes = {
  stockDetails: PropTypes.shape([PropTypes.string, PropTypes.number]).isRequired,
};

export default PrintDetails;
