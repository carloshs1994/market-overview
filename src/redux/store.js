import { createStore, combineReducers, applyMiddleware } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import homeReducer from './home/home';
import detailsReducer from './details/details';

const reducers = combineReducers({
  homeReducer,
  detailsReducer,
});

const store = createStore(
  reducers,
  applyMiddleware(logger, thunk),
);

export default store;
