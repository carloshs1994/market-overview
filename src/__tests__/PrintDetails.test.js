import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import PrintDetails from '../components/PrintDetails';
import store from '../redux/store';

test('matches voice search component snapshot', () => {
  const stock = {
    symbol: 'AAPL',
    name: 'Apple Inc.',
    price: 149.55000000,
    changesPercentage: 0.38934135,
    change: 0.58000183,
    dayLow: 148.75000000,
    dayHigh: 151.41000000,
    yearHigh: 157.26000000,
    yearLow: 103.10000000,
    marketCap: 2472091320320.00000000,
    priceAvg50: 149.04265000,
    priceAvg200: 134.71935000,
    volume: 97750498,
    avgVolume: 78489468,
    exchange: 'NASDAQ',
    open: 150.63000000,
    previousClose: 148.97000000,
    eps: 5.10800000,
    pe: 29.27760500,
    earningsAnnouncement: '2021-07-27T16:30:00.000+0000',
    sharesOutstanding: 16530199400,
    timestamp: 1631575899,
  };
  const tree = renderer.create(
    <Provider store={store}>
      <PrintDetails stockDetails={stock} />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
