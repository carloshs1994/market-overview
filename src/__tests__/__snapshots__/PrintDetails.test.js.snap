// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches voice search component snapshot 1`] = `
Array [
  <li>
    <p>
      symbol
    </p>
    <p>
      AAPL
    </p>
  </li>,
  <li>
    <p>
      name
    </p>
    <p>
      Apple Inc.
    </p>
  </li>,
  <li>
    <p>
      price
    </p>
    <p>
      149.55
    </p>
  </li>,
  <li>
    <p>
      changesPercentage
    </p>
    <p>
      0.38934135
    </p>
  </li>,
  <li>
    <p>
      change
    </p>
    <p>
      0.58000183
    </p>
  </li>,
  <li>
    <p>
      dayLow
    </p>
    <p>
      148.75
    </p>
  </li>,
  <li>
    <p>
      dayHigh
    </p>
    <p>
      151.41
    </p>
  </li>,
  <li>
    <p>
      yearHigh
    </p>
    <p>
      157.26
    </p>
  </li>,
  <li>
    <p>
      yearLow
    </p>
    <p>
      103.1
    </p>
  </li>,
  <li>
    <p>
      marketCap
    </p>
    <p>
      2472091320320
    </p>
  </li>,
  <li>
    <p>
      priceAvg50
    </p>
    <p>
      149.04265
    </p>
  </li>,
  <li>
    <p>
      priceAvg200
    </p>
    <p>
      134.71935
    </p>
  </li>,
  <li>
    <p>
      volume
    </p>
    <p>
      97750498
    </p>
  </li>,
  <li>
    <p>
      avgVolume
    </p>
    <p>
      78489468
    </p>
  </li>,
  <li>
    <p>
      exchange
    </p>
    <p>
      NASDAQ
    </p>
  </li>,
  <li>
    <p>
      open
    </p>
    <p>
      150.63
    </p>
  </li>,
  <li>
    <p>
      previousClose
    </p>
    <p>
      148.97
    </p>
  </li>,
  <li>
    <p>
      eps
    </p>
    <p>
      5.108
    </p>
  </li>,
  <li>
    <p>
      pe
    </p>
    <p>
      29.277605
    </p>
  </li>,
  <li>
    <p>
      earningsAnnouncement
    </p>
    <p>
      2021-07-27T16:30:00.000+0000
    </p>
  </li>,
  <li>
    <p>
      sharesOutstanding
    </p>
    <p>
      16530199400
    </p>
  </li>,
  <li>
    <p>
      timestamp
    </p>
    <p>
      1631575899
    </p>
  </li>,
]
`;
